core/lib/multivm/src/tracers/prestate_tracer/vm_refunds_enhancement/mod.rs

```diff
@@ -0,0 +1,160 @@
+use std::{collections::HashMap, sync::Arc};
+
+use once_cell::sync::OnceCell;
+use zk_evm_1_3_3::tracing::{BeforeExecutionData, VmLocalStateData};
+use zksync_state::{StoragePtr, WriteStorage};
+use zksync_types::{
+    get_code_key, get_nonce_key, web3::signing::keccak256, AccountTreeId, Address, StorageKey,
+    StorageValue, H256, L2_ETH_TOKEN_ADDRESS,
+};
+use zksync_utils::{address_to_h256, h256_to_u256};
+
+use super::{Account, PrestateTracer, State};
+use crate::{
+    interface::dyn_tracers::vm_1_3_3::DynTracer,
+    vm_refunds_enhancement::{BootloaderState, HistoryMode, SimpleMemory, VmTracer, ZkSyncVmState},
+};
+
+impl<S: WriteStorage, H: HistoryMode> DynTracer<S, SimpleMemory<H>> for PrestateTracer {
+    fn before_execution(
```

Comment:
seems you use only modified storage key everywhere, so you can just implement it once and use for all vms